src/components/header/Header.component.tsx

```diff
@@ -46,55 +45,44 @@ export const HeaderComponent: React.FC<HeaderProps> = ({ themeMode, setThemeMode
         setThemeMode('light');
     };
 
-    const showAboutPage = () => {
-        setRoutePath(ROUTES.ABOUT);
-    };
-
-    const showProductsPage = () => {
-        setRoutePath(ROUTES.PRODUCTS);
-    };
+    const lightColorModeIconActiveClass = themeMode === 'dark' ? '' : styles.headerModeActiveIcon;
+    const darkColorModeIconActiveClass = themeMode === 'dark' ? styles.headerModeActiveIcon : '';
 
-    const aboutPageActiveClass: string = route.path === ROUTES.ABOUT ? styles.headerNavLinkActive : '';
-    const productsPageActiveClass: string =
-        route.path === ROUTES.PRODUCTS || route.path.startsWith(ROUTES.PRODUCT) ? styles.headerNavLinkActive : '';
+    const headerNavLinkClasses = (isActive: boolean) => `${styles.headerNavLink} ${isActive ? styles.headerNavLinkActive : ''}`;
 
     return (
         <header className={styles.header} onClick={handleMobileMenuClose}>
             <div className={`${appStyles.contentWrapper} ${styles.headerColumnWrapper}`}>
-                <a className={styles.headerLogoLink} href="/">
+                <Link to={ROUTES.ROOT} className={styles.headerLogoLink}>
                     <LogoIconComponent className={styles.headerLogoIcon} />
-                </a>
+                </Link>
                 <div className={styles.headerMenuWrapper}>
                     <div className={styles.headerModeSwitcher}>
                         <button className={styles.headerModeSwitcherButton} onClick={setLightColorScheme} title="Switch to light mode">
-                            <LightColorModeIconComponent
-                                className={`${styles.headerModeIcon} ${themeMode === 'dark' ? '' : styles.headerModeActiveIcon}`}
-                            />
+                            <LightColorModeIconComponent className={`${styles.headerModeIcon} ${lightColorModeIconActiveClass}`} />
                         </button>
                         <ColorModeDividerIconComponent />
                         <button className={styles.headerModeSwitcherButton} onClick={setDarkColorScheme} title="Switch to dark mode">
-                            <DarkColorModeIconComponent
-                                className={`${styles.headerModeIcon} ${themeMode === 'dark' ? styles.headerModeActiveIcon : ''}`}
-                            />
+                            <DarkColorModeIconComponent className={`${styles.headerModeIcon} ${darkColorModeIconActiveClass}`} />
                         </button>
                     </div>
```

Comment:
You can move this code to separate `ThemeSwitcher` component

Reply:
fixed